src/utils/geometry.js

```diff
@@ -0,0 +1,78 @@
+function clipPolygon(preClippedVertices, polygon, isHole) {
```

Comment:
not sure about the need of having this method inside a utility file I think `clipPolygon` is only used in `feature-decoder.js`